File: sistema-educativo/src/main/java/com/educacion/service/CursoService.java
package com.educacion.service;

import com.educacion.model.Curso;
import com.educacion.repository.CursoRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class CursoService {

    @Autowired
    private CursoRepository cursoRepository;

    public List<Curso> obtenerCursos() {
        return cursoRepository.findAll();
    }

    public ResponseEntity<Curso> obtenerCursoPorId(String id) {
        return cursoRepository.findById(id).map(ResponseEntity::ok)
                .orElse(ResponseEntity.notFound().build());
    }

    public Curso crearCurso(Curso curso) {
        return cursoRepository.save(curso);
    }

    public ResponseEntity<Curso> actualizarCurso(String id, Curso curso) {
        return cursoRepository.findById(id).map(c -> {
            c.setNombre(curso.getNombre());
            c.setDescripcion(curso.getDescripcion());
            return ResponseEntity.ok(cursoRepository.save(c));
        }).orElse(ResponseEntity.notFound().build());
    }

    public ResponseEntity<Object> eliminarCurso(String id) {
        return cursoRepository.findById(id).map(c -> {
            cursoRepository.delete(c);
            return ResponseEntity.ok().build();
        }).orElse(ResponseEntity.notFound().build());
    }
}

File: sistema-educativo-frontend/src/services/cursoService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/cursos';

const cursoService = {
        obtenerCursos: () => axios.get(API_URL),
        obtenerCursoPorId: (id) => axios.get(`${API_URL}/${id}`),
        crearCurso: (curso) => axios.post(API_URL, curso),
        actualizarCurso: (id, curso) => axios.put(`${API_URL}/${id}`, curso),
        eliminarCurso: (id) => axios.delete(`${API_URL}/${id}`)
};

export default cursoService;